The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10497890899/job/29081689548) [(plain)](https://github.com/rust-lang/rust/commit/d7b2fd4213971edd09ed72f9a6541a036d016666/checks/29081689548/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.787 Building wheels for collected packages: reuse
#13 2.788   Building wheel for reuse (pyproject.toml): started
#13 3.030   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.031   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 3.031   Stored in directory: /tmp/pip-ephem-wheel-cache-7_10n22j/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.034 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.419 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.420 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.937 Collecting virtualenv
#13 3.937 Collecting virtualenv
#13 3.996   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 4.171      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 33.1 MB/s eta 0:00:00
#13 4.213 Collecting distlib<1,>=0.3.7
#13 4.219   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 4.230      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 52.9 MB/s eta 0:00:00
#13 4.262 Collecting platformdirs<5,>=3.9.1
#13 4.271   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 4.309 Collecting filelock<4,>=3.12.2
#13 4.315   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 4.395 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.580 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 4.7s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      194496 kB
DirectMap2M:     7145472 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/6b678c57b63b3062fb97130b3617b82657f59c80/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-6b678c57b63b3062fb97130b3617b82657f59c80-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 28.41s
##[endgroup]
fmt check
Diff in /checkout/library/std/src/sys/pal/windows/c.rs:112:
 #[cfg(not(target_vendor = "win7"))]
     target_arch = "x86",
-    link(
-        name = "bcryptprimitives",
-        kind = "raw-dylib",
-        kind = "raw-dylib",
-        import_name_type = "undecorated"
-    )
+    link(name = "bcryptprimitives", kind = "raw-dylib", import_name_type = "undecorated")
-#[cfg_attr(
-#[cfg_attr(
-    not(target_arch = "x86"),
-    link(name = "bcryptprimitives", kind = "raw-dylib")
-)]
+#[cfg_attr(not(target_arch = "x86"), link(name = "bcryptprimitives", kind = "raw-dylib"))]
 extern "system" {
     pub fn ProcessPrng(pbdata: *mut u8, cbdata: usize) -> BOOL;
 }
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/sys/sync/thread_parking/xous.rs" "/checkout/library/std/src/sys/sync/thread_parking/windows7.rs" "/checkout/library/std/src/sys/sync/thread_parking/unsupported.rs" "/checkout/library/std/src/sys/sync/once/queue.rs" "/checkout/library/std/src/sys/sync/once/mod.rs" "/checkout/library/std/src/sys/sync/once/no_threads.rs" "/checkout/library/std/src/sys/sync/once/futex.rs" "/checkout/library/std/src/sys/backtrace.rs" "/checkout/library/std/src/sys/thread_local/mod.rs" "/checkout/library/std/src/sys/thread_local/native/lazy.rs" "/checkout/library/std/src/sys/thread_local/native/mod.rs" "/checkout/library/std/src/sys/thread_local/native/eager.rs" "/checkout/library/std/src/sys/thread_local/key/sgx.rs" "/checkout/library/std/src/sys/thread_local/key/unix.rs" "/checkout/library/std/src/sys/thread_local/key/xous.rs" "/checkout/library/std/src/sys/thread_local/key/racy.rs" "/checkout/library/std/src/sys/thread_local/key/tests.rs" "/checkout/library/std/src/sys/thread_local/key/windows.rs" "/checkout/library/std/src/sys/thread_local/os.rs" "/checkout/library/std/src/sys/thread_local/statik.rs" "/checkout/library/std/src/sys/thread_local/destructors/list.rs" "/checkout/library/std/src/sys/thread_local/destructors/linux_like.rs" "/checkout/library/std/src/sys/thread_local/guard/key.rs" "/checkout/library/std/src/sys/thread_local/guard/solid.rs" "/checkout/library/std/src/sys/thread_local/guard/apple.rs" "/checkout/library/std/src/sys/thread_local/guard/windows.rs" "/checkout/library/std/src/sys/os_str/bytes/tests.rs" "/checkout/library/std/src/sys/os_str/mod.rs" "/checkout/library/std/src/sys/os_str/bytes.rs" "/checkout/library/std/src/sys/os_str/wtf8.rs" "/checkout/library/std/src/sys/mod.rs" "/checkout/library/std/src/sys/anonymous_pipe/mod.rs" "/checkout/library/std/src/sys/anonymous_pipe/unix.rs" "/checkout/library/std/src/sys/anonymous_pipe/unsupported.rs" "/checkout/library/std/src/sys/anonymous_pipe/windows.rs" "/checkout/library/std/src/sys/pal/windows/time.rs" "/checkout/library/std/src/sys/pal/windows/pipe.rs" "/checkout/library/std/src/sys/pal/windows/alloc/tests.rs" "/checkout/library/std/src/sys/pal/windows/stdio/tests.rs" "/checkout/library/std/src/sys/pal/windows/stack_overflow_uwp.rs" "/checkout/library/std/src/sys/pal/windows/c/windows_sys.rs" "/checkout/library/std/src/sys/pal/windows/handle/tests.rs" "/checkout/library/std/src/sys/pal/windows/api.rs" "/checkout/library/std/src/sys/pal/windows/mod.rs" "/checkout/library/std/src/sys/pal/windows/c.rs" "/checkout/library/std/src/sys/pal/windows/thread.rs" "/checkout/library/std/src/sys/pal/windows/rand.rs" "/checkout/library/std/src/sys/pal/windows/args.rs" "/checkout/library/std/src/sys/pal/windows/handle.rs" "/checkout/library/std/src/sys/pal/windows/futex.rs" "/checkout/library/std/src/sys/pal/windows/alloc.rs" "/checkout/library/std/src/sys/pal/windows/net.rs" "/checkout/library/std/src/sys/pal/windows/stack_overflow.rs" "/checkout/library/std/src/sys/pal/windows/api/tests.rs" "/checkout/library/std/src/sys/pal/windows/process.rs" "/checkout/library/std/src/sys/pal/windows/os.rs" "/checkout/library/std/src/sys/pal/windows/env.rs" "/checkout/library/std/src/sys/pal/windows/process/tests.rs" "/checkout/library/std/src/sys/pal/windows/io.rs" "/checkout/library/std/src/sys/pal/windows/stdio.rs" "/checkout/library/std/src/sys/pal/windows/os/tests.rs" "/checkout/library/std/src/sys/pal/windows/args/tests.rs" "/checkout/library/std/src/sys/pal/windows/compat.rs" "/checkout/library/std/src/sys/sync/thread_parking/pthread.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Wed Aug 21 21:45:06 UTC 2024
  network time: Wed, 21 Aug 2024 21:45:06 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/std/src/sys/pal/windows/c.rs
+++ b/library/std/src/sys/pal/windows/c.rs
@@ -112,16 +112,9 @@ if #[cfg(not(target_vendor = "uwp"))] {
 #[cfg(not(target_vendor = "win7"))]
 #[cfg_attr(
     target_arch = "x86",
-    link(
-        name = "bcryptprimitives",
-        kind = "raw-dylib",
-        import_name_type = "undecorated"
-    )
+    link(name = "bcryptprimitives", kind = "raw-dylib", import_name_type = "undecorated")
 )]
-#[cfg_attr(
-    not(target_arch = "x86"),
-    link(name = "bcryptprimitives", kind = "raw-dylib")
-)]
+#[cfg_attr(not(target_arch = "x86"), link(name = "bcryptprimitives", kind = "raw-dylib"))]
 extern "system" {
     pub fn ProcessPrng(pbdata: *mut u8, cbdata: usize) -> BOOL;
 }

